src/fparser/two/tests/test_parser.py

```diff
@@ -84,3 +84,47 @@ class do not affect current calls.
     with pytest.raises(ValueError) as excinfo:
         parser = ParserFactory().create(std="invalid")
         assert "is an invalid standard" in str(excinfo.value)
+
+
+def test_deepcopy():
+    """
+    Test that we can deepcopy a parsed fparser tree.
+    """
+
+    f90_source = """
+program main
+
+  implicit none
+
+end
+"""
+    parser = ParserFactory().create(std="f2008")
+    reader = FortranStringReader(f90_source)
+    ast = parser(reader)
+
+    import copy
+
+    _ = copy.deepcopy(ast)
+
+
+def test_pickle():
+    """
+    Test that we can pickle and unpickle a parsed fparser tree.
+    """
+
+    f90_source = """
+program main
+
+  implicit none
+
+end
+"""
+
+    parser = ParserFactory().create(std="f2008")
+    reader = FortranStringReader(f90_source)
+    ast = parser(reader)
+
+    import pickle
+
+    s = pickle.dumps(ast)
+    _ = pickle.loads(s)
```

Comment:
Please add some minimal check here that the unpickled parse tree has the right structure. e.g. that its root node is of the same type as the root node of `ast` but is a different object.

Reply:
Okidoki. It's now a test parsing pairwise both trees.